### DISCUSSION TITLE
Partial mux tracing

### DISCUSSION THREAD

**Discussion by appiepollo14** [open-telemetry/opentelemetry-go#5619] [Q&A] [answered]
Hi,

In [this](https://github.com/open-telemetry/opentelemetry-go/blob/main/example/dice/main.go#L84) beautiful example, the whole mux server is being traced. I'm trying to find out how to do partial tracing, meaning, selected endpoints to be traced and not /health / ready. Does anyone have a gist or an example how to do this with mux? 

Kind regards, Asjer

**Comment by dmathieu** [comment, marked as answer]
You can do this using the [WithFilter](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp#WithFilter) option.
Something like this:

```golang
handler := otelhttp.NewHandler(mux, "/", otelhttp.WithFilter(func(r *http.Request) bool {
  if (r.URL.Path == "/health" || r.URL.Path == "/ready") {
    return false
  }

  return true
})
```

**Reply by appiepollo14** [reply]
Perfect, thank you!
